### DISCUSSION TITLE
Cannot deploy contact from a contract (Storage Factory)

### DISCUSSION THREAD

**Discussion by ggeorges11** [Cyfrin/foundry-full-course-cu#3025] [Q&A]
Hello, I am at the Storage Factory Lesson 3 --> Deploying a contract from a contract. After writing my contract StorageFactory code in Remix with Patrick instructions, i compiled it and everything looked good but I could not deploy it. I suspect this has to do with the environment but i don't know how to fix this. Please help! Thank you.
![image](https://github.com/user-attachments/assets/136568e0-4419-4b61-a854-d12604898f08)

**Comment by ggeorges11** [comment]
I found out the answer. I was on the zksync layer 2 network and apparently this didn't work for me. I had to go back to the ETH Sepolia testnet for the deployment to work.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.

**Comment by ggeorges11** [comment]
Thank you so much!!

From: EngrPips ***@***.***>
Date: Wednesday, December 11, 2024 at 4:00 AM
To: Cyfrin/foundry-full-course-cu ***@***.***>
Cc: ggeorges11 ***@***.***>, Author ***@***.***>
Subject: Re: [Cyfrin/foundry-full-course-cu] Cannot deploy contact from a contract (Storage Factory) (Discussion #3025)

Awesome, I am glad you figured it out. Keep crushing it.

**Comment by Joezzyongit** [comment]
i can't deploy my contract to the sepolia testnet because the fee is too high too bad for me

**Reply by EngrPips** [reply]
Hey @Joezzyongit, running on sepolia has become expensive lately If you can't afford it. I will advise you to test things locally.
